### PR TITLE
Move e2e-image and test-image into test-infra

### PR DESCRIPTION
**Pull request by fejta**
Also update kubekins-e2e to the 20160824 version.

For https://github.com/kubernetes/kubernetes/issues/31149

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/444)

### CODE REVIEW

File: jenkins/e2e-image/e2e-runner.sh [hunk 1]
```diff
@@ -0,0 +1,418 @@
+#!/bin/bash
```
Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: pull https://github.com/kubernetes/kubernetes/pull/31330/commits/12ea9cb1db4bdd71857fc16d3293da37fe98c29a into this?

Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: Done

File: jenkins/dockerized-e2e-runner.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ mkdir -p "${HOST_ARTIFACTS_DIR}"
 : ${JENKINS_GCE_SSH_PRIVATE_KEY_FILE:='/var/lib/jenkins/gce_keys/google_compute_engine'}
 : ${JENKINS_GCE_SSH_PUBLIC_KEY_FILE:='/var/lib/jenkins/gce_keys/google_compute_engine.pub'}
 
-KUBEKINS_E2E_IMAGE_TAG='v20160822'
+KUBEKINS_E2E_IMAGE_TAG='v20160824'
```
Comment (ixdy) [inline jenkins/dockerized-e2e-runner.sh]: the commit description is technically out of date now

File: jenkins/e2e-image/Dockerfile [hunk 3]
```diff
@@ -43,6 +43,10 @@ ENV FAIL_ON_GCP_RESOURCE_LEAK=true \
 # JENKINS_AWS_SSH_PUBLIC_KEY_FILE
 # JENKINS_AWS_CREDENTIALS_FILE
 
-ADD ["sh2ju.sh", "e2e-runner.sh", "e2e.go", "upload-to-gcs.sh", "${WORKSPACE}/"]
+ADD ["e2e-runner.sh", \
+    "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/e2e.go", \
+    "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/upload-to-gcs.sh", \
+    "https://raw.githubusercontent.com/kubernetes/kubernetes/master/third_party/forked/shell2junit/sh2ju.sh", \
+    "/workspace/"]
```
Comment (ixdy) [inline jenkins/e2e-image/Dockerfile]: either keep ${WORKSPACE} or change the entrypoint?

**Comment by ixdy** [comment]
basically LG besides a few nits

File: jenkins/e2e-image/Dockerfile (hunk 3 above)
Comment (fejta) [inline jenkins/e2e-image/Dockerfile]: Done

File: jenkins/dockerized-e2e-runner.sh (hunk 2 above)
Comment (fejta) [inline jenkins/dockerized-e2e-runner.sh]: I even fixed it! But apparently reword doesn't work as I expect

**Comment by ixdy** [comment]
LGTM!

File: jenkins/e2e-image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# GOOGLE_APPLICATION_CREDENTIALS
+# JENKINS_GCE_SSH_PRIVATE_KEY_FILE
+# JENKINS_GCE_SSH_PUBLIC_KEY_FILE
+# JENKINS_AWS_SSH_PRIVATE_KEY_FILE
+# JENKINS_AWS_SSH_PUBLIC_KEY_FILE
+# JENKINS_AWS_CREDENTIALS_FILE
+
+ADD ["e2e-runner.sh", \
+    "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/e2e.go", \
+    "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/jenkins/upload-to-gcs.sh", \
```
Comment (ixdy) [inline jenkins/e2e-image/Dockerfile]: apparently URLs are not made executable.

Comment (fejta) [inline jenkins/e2e-image/Dockerfile]: https://github.com/kubernetes/test-infra/pull/450

**Pull request merged**